### ISSUE TITLE
feat(actions): uniqueSet() for transaction capability

### ISSUE THREAD

**Issue by prescottprue** [prescottprue/redux-firestore#208]
**What is the feature?**
Exposing a uniqueSet() in props.firestore that is similar to the one within props.firebase would be fantastic for use cases where multiple clients are writing to the same document.

Originally requested in [419 of react-redux-firebase](https://github.com/prescottprue/react-redux-firebase/issues/419) by @giles-t

**What version would this apply to?**

**Do you have thoughts/examples on how the feature would be used (i.e. API)?**

`uniqueSet` would match what react-redux-firebase does for RTDB, so that makes the most sense

**Comment by illuminist** [comment]
```js
const uniqueSet = async (docRef, value, onComplete) => {
  await firebase.runTransaction(
    async transaction => {
      const docSnap = await transaction.get(docRef)
      if (docSnap.exists) {
        const newError = new Error('Document already exists.')
        if (onComplete) onComplete(newError)
        throw newError
      }
      return transaction.set(value)
    }
  )
  if (onComplete) onComplete()
}
```
What do you think?

**Comment by prescottprue** [comment]
Yup, that is the desired behavior @illuminist, thanks for posting. Feel free to make a PR with the changes, and we can get them in 😄

**Comment by illuminist** [comment]
Does it need to use `redux-firestore` function? Somehow I feel like I need to make it accept `QueryOption` like the other functions.

**Comment by prescottprue** [comment]
@illuminist yeah, it would live in redux-firestore since that is where the other Firestore specific actions live. Accepting QueryOption like you mentioned would be good to match the other call signatures
